{"matches":[{"vulnerability":{"id":"GHSA-2jv5-9r88-3w3p","dataSource":"https://github.com/advisories/GHSA-2jv5-9r88-3w3p","namespace":"github:language:python","severity":"High","urls":[],"description":"python-multipart vulnerable to Content-Type Header ReDoS","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-24762","epss":0.01802,"percentile":0.81955,"date":"2025-06-03"}],"fix":{"versions":["0.0.7"],"state":"fixed"},"advisories":[],"risk":1.3515000000000001},"relatedVulnerabilities":[{"id":"CVE-2024-24762","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-24762","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4","https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p","https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74","https://github.com/encode/starlette/commit/13e5c26a27f4903924624736abd6131b2da80cc5","https://github.com/encode/starlette/security/advisories/GHSA-93gm-qmq6-w238","https://github.com/tiangolo/fastapi/commit/9d34ad0ee8a0dfbbcce06f76c2d5d851085024fc","https://github.com/tiangolo/fastapi/releases/tag/0.109.1","https://github.com/tiangolo/fastapi/security/advisories/GHSA-qf9m-vfgh-m389","https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4","https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p","https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74","https://github.com/encode/starlette/commit/13e5c26a27f4903924624736abd6131b2da80cc5","https://github.com/encode/starlette/security/advisories/GHSA-93gm-qmq6-w238","https://github.com/tiangolo/fastapi/commit/9d34ad0ee8a0dfbbcce06f76c2d5d851085024fc","https://github.com/tiangolo/fastapi/releases/tag/0.109.1","https://github.com/tiangolo/fastapi/security/advisories/GHSA-qf9m-vfgh-m389"],"description":"`python-multipart` is a streaming multipart parser for Python. When using form data, `python-multipart` uses a Regular Expression to parse the HTTP `Content-Type` header, including options. An attacker could send a custom-made `Content-Type` option that is very difficult for the RegEx to process, consuming CPU resources and stalling indefinitely (minutes or more) while holding the main event loop. This means that process can't handle any more requests, leading to regular expression denial of service. This vulnerability has been patched in version 0.0.7.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-24762","epss":0.01802,"percentile":0.81955,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"python-multipart","version":"0.0.5"}},"found":{"versionConstraint":"<=0.0.6 (python)","vulnerabilityID":"GHSA-2jv5-9r88-3w3p"},"fix":{"suggestedVersion":"0.0.7"}}],"artifact":{"id":"ed67ce599783b6ad","name":"python-multipart","version":"0.0.5","type":"python","locations":[{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache"],"cpes":["cpe:2.3:a:fastapiexpert:python-multipart:0.0.5:*:*:*:*:python:*:*"],"purl":"pkg:pypi/python-multipart@0.0.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-v5gw-mw7f-84px","dataSource":"https://github.com/advisories/GHSA-v5gw-mw7f-84px","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Starlette has Path Traversal vulnerability in StaticFiles","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:N","metrics":{"baseScore":3.7,"exploitabilityScore":2.3,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-29159","epss":0.01332,"percentile":0.7903,"date":"2025-06-03"}],"fix":{"versions":["0.27.0"],"state":"fixed"},"advisories":[],"risk":0.57942},"relatedVulnerabilities":[{"id":"CVE-2023-29159","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-29159","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/encode/starlette/releases/tag/0.27.0","https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px","https://jvn.jp/en/jp/JVN95981715/","https://github.com/encode/starlette/releases/tag/0.27.0","https://github.com/encode/starlette/security/advisories/GHSA-v5gw-mw7f-84px","https://jvn.jp/en/jp/JVN95981715/"],"description":"Directory traversal vulnerability in Starlette versions 0.13.5 and later and prior to 0.27.0 allows a remote unauthenticated attacker to view files in a web service which was built using Starlette.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-29159","epss":0.01332,"percentile":0.7903,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"starlette","version":"0.17.1"}},"found":{"versionConstraint":">=0.13.5,<0.27.0 (python)","vulnerabilityID":"GHSA-v5gw-mw7f-84px"},"fix":{"suggestedVersion":"0.27.0"}}],"artifact":{"id":"a54ad58555e0bb13","name":"starlette","version":"0.17.1","type":"python","locations":[{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:encode:starlette:0.17.1:*:*:*:*:python:*:*"],"purl":"pkg:pypi/starlette@0.17.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-74m5-2c7w-9w3x","dataSource":"https://github.com/advisories/GHSA-74m5-2c7w-9w3x","namespace":"github:language:python","severity":"High","urls":[],"description":"MultipartParser denial of service with too many fields or files","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30798","epss":0.00726,"percentile":0.71543,"date":"2025-06-03"}],"fix":{"versions":["0.25.0"],"state":"fixed"},"advisories":[],"risk":0.5445},"relatedVulnerabilities":[{"id":"CVE-2023-30798","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2023-30798","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/encode/starlette/commit/8c74c2c8dba7030154f8af18e016136bea1938fa","https://github.com/encode/starlette/security/advisories/GHSA-74m5-2c7w-9w3x","https://vulncheck.com/advisories/starlette-multipartparser-dos","https://github.com/encode/starlette/commit/8c74c2c8dba7030154f8af18e016136bea1938fa","https://github.com/encode/starlette/security/advisories/GHSA-74m5-2c7w-9w3x","https://vulncheck.com/advisories/starlette-multipartparser-dos"],"description":"There MultipartParser usage in Encode's Starlette python framework before versions 0.25.0 allows an unauthenticated and remote attacker to specify any number of form fields or files which can cause excessive memory usage resulting in denial of service of the HTTP service.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"disclosure@vulncheck.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2023-30798","epss":0.00726,"percentile":0.71543,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"starlette","version":"0.17.1"}},"found":{"versionConstraint":"<0.25.0 (python)","vulnerabilityID":"GHSA-74m5-2c7w-9w3x"},"fix":{"suggestedVersion":"0.25.0"}}],"artifact":{"id":"a54ad58555e0bb13","name":"starlette","version":"0.17.1","type":"python","locations":[{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:encode:starlette:0.17.1:*:*:*:*:python:*:*"],"purl":"pkg:pypi/starlette@0.17.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"69.0.3"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"f7cf308e954ed2ba","name":"setuptools","version":"69.0.3","type":"python","locations":[{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:69.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@69.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-59g5-xgcq-4qw3","dataSource":"https://github.com/advisories/GHSA-59g5-xgcq-4qw3","namespace":"github:language:python","severity":"High","urls":[],"description":"Denial of service (DoS) via deformation `multipart/form-data` boundary","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-53981","epss":0.00379,"percentile":0.58682,"date":"2025-06-03"}],"fix":{"versions":["0.0.18"],"state":"fixed"},"advisories":[],"risk":0.28425},"relatedVulnerabilities":[{"id":"CVE-2024-53981","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-53981","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/Kludex/python-multipart/commit/c4fe4d3cebc08c660e57dd709af1ffa7059b3177","https://github.com/Kludex/python-multipart/security/advisories/GHSA-59g5-xgcq-4qw3"],"description":"python-multipart is a streaming multipart parser for Python. When parsing form data, python-multipart skips line breaks (CR \\r or LF \\n) in front of the first boundary and any tailing bytes after the last boundary. This happens one byte at a time and emits a log event each time, which may cause excessive logging for certain inputs. An attacker could abuse this by sending a malicious request with lots of data before the first or after the last boundary, causing high CPU load and stalling the processing thread for a significant amount of time. In case of ASGI application, this could stall the event loop and prevent other requests from being processed, resulting in a denial of service (DoS). This vulnerability is fixed in 0.0.18.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-53981","epss":0.00379,"percentile":0.58682,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"python-multipart","version":"0.0.5"}},"found":{"versionConstraint":"<0.0.18 (python)","vulnerabilityID":"GHSA-59g5-xgcq-4qw3"},"fix":{"suggestedVersion":"0.0.18"}}],"artifact":{"id":"ed67ce599783b6ad","name":"python-multipart","version":"0.0.5","type":"python","locations":[{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/python_multipart-0.0.5.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["Apache"],"cpes":["cpe:2.3:a:fastapiexpert:python-multipart:0.0.5:*:*:*:*:python:*:*"],"purl":"pkg:pypi/python-multipart@0.0.5","upstreams":[]}},{"vulnerability":{"id":"GHSA-jjg7-2v4v-x38h","dataSource":"https://github.com/advisories/GHSA-jjg7-2v4v-x38h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62113,"date":"2025-06-03"}],"fix":{"versions":["3.7"],"state":"fixed"},"advisories":[],"risk":0.24416000000000004},"relatedVulnerabilities":[{"id":"CVE-2024-3651","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3651","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb","https://github.com/kjd/idna/commit/1d365e17e10d72d0b7876316fc7b9ca0eebdd38d","https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb"],"description":"A vulnerability was identified in the kjd/idna library, specifically within the `idna.encode()` function, affecting version 3.6. The issue arises from the function's handling of crafted input strings, which can lead to quadratic complexity and consequently, a denial of service condition. This vulnerability is triggered by a crafted input that causes the `idna.encode()` function to process the input with considerable computational load, significantly increasing the processing time in a quadratic manner relative to the input size.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":7.5,"exploitabilityScore":3.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":6.2,"exploitabilityScore":2.6,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3651","epss":0.00436,"percentile":0.62113,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"idna","version":"3.6"}},"found":{"versionConstraint":"<3.7 (python)","vulnerabilityID":"GHSA-jjg7-2v4v-x38h"},"fix":{"suggestedVersion":"3.7"}}],"artifact":{"id":"1b4c65e6cbd10dc2","name":"idna","version":"3.6","type":"python","locations":[{"path":"/lib/python3.8/site-packages/idna-3.6.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/idna-3.6.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/idna-3.6.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/idna-3.6.dist-info/RECORD","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:kim_davies_<kim+pypi_project:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypi_project:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypiproject:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypiproject:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypi_project:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim+pypi:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim+pypi:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypi:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypi:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypiproject:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim-davies-<kim+pypi:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:kim_davies_<kim+pypi:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:idna:python-idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:idna:python_idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python-idna:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python_idna:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:python:idna:3.6:*:*:*:*:*:*:*","cpe:2.3:a:idna:idna:3.6:*:*:*:*:*:*:*"],"purl":"pkg:pypi/idna@3.6","upstreams":[]}},{"vulnerability":{"id":"GHSA-cx63-2mw6-8hw5","dataSource":"https://github.com/advisories/GHSA-cx63-2mw6-8hw5","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools vulnerable to Command Injection via package URL","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}],"fix":{"versions":["70.0.0"],"state":"fixed"},"advisories":[],"risk":0.185005},"relatedVulnerabilities":[{"id":"CVE-2024-6345","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-6345","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5","https://github.com/pypa/setuptools/commit/88807c7062788254f654ea8c03427adc859321f0","https://huntr.com/bounties/d6362117-ad57-4e83-951f-b8141c6e7ca5"],"description":"A vulnerability in the package_index module of pypa/setuptools versions up to 69.1.1 allows for remote code execution via its download functions. These functions, which are used to download packages from URLs provided by users or retrieved from package index servers, are susceptible to code injection. If these functions are exposed to user-controlled inputs, such as package URLs, they can execute arbitrary commands on the system. The issue is fixed in version 70.0.","cvss":[{"source":"security@huntr.dev","type":"Secondary","version":"3.0","vector":"CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-6345","epss":0.00227,"percentile":0.45699,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"69.0.3"}},"found":{"versionConstraint":"<70.0.0 (python)","vulnerabilityID":"GHSA-cx63-2mw6-8hw5"},"fix":{"suggestedVersion":"70.0.0"}}],"artifact":{"id":"f7cf308e954ed2ba","name":"setuptools","version":"69.0.3","type":"python","locations":[{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/setuptools-69.0.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:69.0.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@69.0.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-mr82-8j83-vxmv","dataSource":"https://github.com/advisories/GHSA-mr82-8j83-vxmv","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Pydantic regular expression denial of service","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3772","epss":0.0028,"percentile":0.51263,"date":"2025-06-03"}],"fix":{"versions":["1.10.13"],"state":"fixed"},"advisories":[],"risk":0.1526},"relatedVulnerabilities":[{"id":"CVE-2024-3772","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-3772","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pydantic/pydantic/pull/7360","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6JBZLMSH4GAZOVBMT2JUO2LXHY7M2ALI/","https://github.com/pydantic/pydantic/pull/7360","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6JBZLMSH4GAZOVBMT2JUO2LXHY7M2ALI/"],"description":"Regular expression denial of service in Pydanic < 2.4.0, < 1.10.13 allows remote attackers to cause denial of service via a crafted email string.","cvss":[{"source":"430a6cef-dc26-47e3-9fa8-52fb7f19644e","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.9,"exploitabilityScore":2.3,"impactScore":3.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-3772","epss":0.0028,"percentile":0.51263,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pydantic","version":"1.9.0"}},"found":{"versionConstraint":"<1.10.13 (python)","vulnerabilityID":"GHSA-mr82-8j83-vxmv"},"fix":{"suggestedVersion":"1.10.13"}}],"artifact":{"id":"414e9be36f07392b","name":"pydantic","version":"1.9.0","type":"python","locations":[{"path":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/pydantic-1.9.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:samuel_colvin_project:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvin_project:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvinproject:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvinproject:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pydantic:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pydantic:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pydantic:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pydantic:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvin_project:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvin:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvin:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvinproject:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s_project:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s_project:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:pydantic:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:pydantic:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pydantic:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pydantic:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:sproject:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:sproject:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:samuel_colvin:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s_project:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:pydantic:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s:python-pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s:python_pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:sproject:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pydantic:1.9.0:*:*:*:*:*:*:*","cpe:2.3:a:s:pydantic:1.9.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pydantic@1.9.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-f96h-pmfr-66vw","dataSource":"https://github.com/advisories/GHSA-f96h-pmfr-66vw","namespace":"github:language:python","severity":"High","urls":[],"description":"Starlette Denial of service (DoS) via multipart/form-data","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N","metrics":{"baseScore":0,"exploitabilityScore":3.9,"impactScore":0},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-47874","epss":0.00104,"percentile":0.29644,"date":"2025-06-03"}],"fix":{"versions":["0.40.0"],"state":"fixed"},"advisories":[],"risk":0.07799999999999999},"relatedVulnerabilities":[{"id":"CVE-2024-47874","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-47874","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/encode/starlette/commit/fd038f3070c302bff17ef7d173dbb0b007617733","https://github.com/encode/starlette/security/advisories/GHSA-f96h-pmfr-66vw"],"description":"Starlette is an Asynchronous Server Gateway Interface (ASGI) framework/toolkit. Prior to version 0.40.0, Starlette treats `multipart/form-data` parts without a `filename` as text form fields and buffers those in byte strings with no size limit. This allows an attacker to upload arbitrary large form fields and cause Starlette to both slow down significantly due to excessive memory allocations and copy operations, and also consume more and more memory until the server starts swapping and grinds to a halt, or the OS terminates the server process with an OOM error. Uploading multiple such requests in parallel may be enough to render a service practically unusable, even if reasonable request size limits are enforced by a reverse proxy in front of Starlette. This Denial of service (DoS) vulnerability affects all applications built with Starlette (or FastAPI) accepting form requests. Verison 0.40.0 fixes this issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:N/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":8.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-47874","epss":0.00104,"percentile":0.29644,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"starlette","version":"0.17.1"}},"found":{"versionConstraint":"<0.40.0 (python)","vulnerabilityID":"GHSA-f96h-pmfr-66vw"},"fix":{"suggestedVersion":"0.40.0"}}],"artifact":{"id":"a54ad58555e0bb13","name":"starlette","version":"0.17.1","type":"python","locations":[{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/starlette-0.17.1.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD"],"cpes":["cpe:2.3:a:encode:starlette:0.17.1:*:*:*:*:python:*:*"],"purl":"pkg:pypi/starlette@0.17.1","upstreams":[]}},{"vulnerability":{"id":"GHSA-vqfr-h8mv-ghfj","dataSource":"https://github.com/advisories/GHSA-vqfr-h8mv-ghfj","namespace":"github:language:python","severity":"Critical","urls":[],"description":"h11 accepts some malformed Chunked-Encoding bodies","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-43859","epss":0.00044,"percentile":0.13105,"date":"2025-06-03"}],"fix":{"versions":["0.16.0"],"state":"fixed"},"advisories":[],"risk":0.03982},"relatedVulnerabilities":[{"id":"CVE-2025-43859","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-43859","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/python-hyper/h11/commit/114803a29ce50116dc47951c690ad4892b1a36ed","https://github.com/python-hyper/h11/security/advisories/GHSA-vqfr-h8mv-ghfj"],"description":"h11 is a Python implementation of HTTP/1.1. Prior to version 0.16.0, a leniency in h11's parsing of line terminators in chunked-coding message bodies can lead to request smuggling vulnerabilities under certain conditions. This issue has been patched in version 0.16.0. Since exploitation requires the combination of buggy h11 with a buggy (reverse) proxy, fixing either component is sufficient to mitigate this issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:N","metrics":{"baseScore":9.1,"exploitabilityScore":3.9,"impactScore":5.2},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-43859","epss":0.00044,"percentile":0.13105,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"h11","version":"0.14.0"}},"found":{"versionConstraint":"<0.16.0 (python)","vulnerabilityID":"GHSA-vqfr-h8mv-ghfj"},"fix":{"suggestedVersion":"0.16.0"}}],"artifact":{"id":"f102630f1875bffe","name":"h11","version":"0.14.0","type":"python","locations":[{"path":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/METADATA","accessPath":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/RECORD","accessPath":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/top_level.txt","accessPath":"/lib/python3.8/site-packages/h11-0.14.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:nathaniel_j__smith_project:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith_project:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith_project:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smithproject:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:nathaniel_j__smith:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs_project:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:python-h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:python_h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njsproject:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python-h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python_h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:python:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:h11:h11:0.14.0:*:*:*:*:*:*:*","cpe:2.3:a:njs:h11:0.14.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/h11@0.14.0","upstreams":[]}}],"source":{"type":"sbom-file","target":"nebula-lighthouse-service-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/nebula-lighthouse-service-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/nebula-lighthouse-service-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-04T01:32:23Z_1749035557.tar.zst?checksum=sha256%3A77238f4ffbc7cf72bf4dedabbb21a3d0dd3ade90fb5d9df7c4da2870f8d1a9b7","built":"2025-06-04T11:12:37Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-04T01:32:23Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-04T01:32:32Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-04T01:32:33Z","input":"xxh64:03c9c4d182d77791"},"debian":{"captured":"2025-06-04T01:32:49Z","input":"xxh64:169ceb8fbcb8d6f1"},"epss":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:bf87c4c234d02346"},"github":{"captured":"2025-06-04T01:32:30Z","input":"xxh64:3dab0be50635e057"},"kev":{"captured":"2025-06-04T01:32:45Z","input":"xxh64:a934e75f468f4e09"},"mariner":{"captured":"2025-06-04T01:32:43Z","input":"xxh64:ac698752085c186f"},"nvd":{"captured":"2025-06-04T01:33:28Z","input":"xxh64:389887843a59d29c"},"oracle":{"captured":"2025-06-04T01:32:35Z","input":"xxh64:0b8aabcfb6505b62"},"rhel":{"captured":"2025-06-04T04:49:10Z","input":"xxh64:947d0caeb19e9ccc"},"sles":{"captured":"2025-06-04T01:32:34Z","input":"xxh64:1c50b1219296628a"},"ubuntu":{"captured":"2025-06-04T01:33:14Z","input":"xxh64:71e22d5c5bbe7d14"},"wolfi":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:e68fca5a870c6769"}}},"timestamp":"2025-06-04T19:33:33.833302+01:00"}}
